tests/Formatter/SmartJsonFormatterTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+namespace Fusions\Test\Monolog\LogDna\Formatter;
+
+use Fusions\Monolog\LogDna\Formatter\SmartJsonFormatter;
+use Fusions\Test\Monolog\LogDna\TestHelperTrait;
+use Monolog\Logger;
+use PHPUnit\Framework\TestCase;
+
+class SmartJsonFormatterTest extends TestCase
+{
+    use TestHelperTrait;
+
+    public function test_format(): void
+    {
+        $record = $this->getRecord(Logger::INFO, 'This is a test message', [
+            'exception' => $this->getExceptionWithStackTrace('This is a test exception', 42, null),
+        ]);
+
+        $formatter = new SmartJsonFormatter;
+        $output    = json_decode($formatter->format($record), true);
+
+        // Core fields.
+        $this->assertSame($record['datetime']->getTimestamp(), $output['lines'][0]['timestamp']);
```

Comment:
Is there a reason we can't just use a fixture file to validate all this json?

Reply:
@liamkeily The initial issue I had was timestamps not matching up, but I've fixed the timestamp in the test log record we use and it works, so all the JSON formatting is tested via fixtures.